gms/gossiper.cc

```diff
@@ -711,7 +711,7 @@ future<> gossiper::do_status_check() {
             continue;
         }
         auto& ep_state = *eps;
-        bool is_alive = ep_state.is_alive();
+        bool is_dead = is_dead_state(ep_state);
```

Comment:
I don't get this change.

`is_dead_state` is not the negation of `is_alive`, a node could be dead but not in DEAD_STATES. So I don't think it's correct to use `is_dead` in place of where `!is_alive` was used earlier (like below with the `expire_time` check)

Shouldn't we use `!is_alive(endpoint)`?

Reply:
Fair point.  I'll change to `is_alive = _live_endpoints.contains(endpoint)` which is the equivalent after this series of what `ep_state.is_alive()` was.